src/passes/OptimizeInstructions.cpp

```diff
@@ -1677,6 +1701,57 @@ struct OptimizeInstructions
           }
         }
       }
+      // signed(x - y) <=> 0  =>  x <=> y
+      //
+      // unsigned(x - y) > 0    =>   x != y
+      // unsigned(x - y) <= 0   =>   x == y
+      // unsigned(x - y) >= 0   =>   i32(1)
+      // unsigned(x - y) < 0    =>   i32(0)
```

Comment:
These two look like they should be handled by more general rules:

```
unsigned(x) >= 0 => i32(1)
unsigned(x) < 0 => i32(0)
```

Also, would be possible to write all of these using matchers? That would make them easier to read.

Reply:
Good catch! Done!